Services/Administration/OJS.Services.Administration.Business/Implementations/ContestResultsAggregatorService.cs

```diff
@@ -1,44 +1,49 @@
 namespace OJS.Services.Administration.Business.Implementations;
 
+using OJS.Services.Ui.Models.Contests;
+using OJS.Services.Ui.Data;
+using OJS.Data.Models.Participants;
+using X.PagedList;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
 using OJS.Data.Models.Contests;
 using OJS.Services.Common;
 using OJS.Services.Common.Models.Contests;
-using OJS.Services.Common.Data;
 using OJS.Services.Common.Models.Contests.Results;
 using SoftUni.AutoMapper.Infrastructure.Extensions;
 using System.Linq;
 
 public class ContestResultsAggregatorService : IContestResultsAggregatorService
 {
-    private readonly IParticipantsCommonDataService participantsCommonData;
     private readonly IContestsActivityService activityService;
+    private readonly IParticipantScoresDataService participantScoresDataService;
+    private readonly IParticipantsDataService participantsData;
 
-    public ContestResultsAggregatorService(IParticipantsCommonDataService participantsCommonData, IContestsActivityService activityService)
+    public ContestResultsAggregatorService(
+        IContestsActivityService activityService,
+        IParticipantScoresDataService participantScoresDataService,
+        IParticipantsDataService participantsData)
     {
-        this.participantsCommonData = participantsCommonData;
         this.activityService = activityService;
+        this.participantScoresDataService = participantScoresDataService;
+        this.participantsData = participantsData;
     }
 
-    public ContestResultsViewModel GetContestResults(
-            Contest contest,
-            bool official,
-            bool isUserAdminOrLecturer,
-            bool isFullResults,
-            bool isExportResults = false)
+    public ContestResultsViewModel GetContestResults(ContestResultsModel contestResultsModel)
     {
         var contestActivityEntity = this.activityService
-            .GetContestActivity(contest.Map<ContestForActivityServiceModel>());
+            .GetContestActivity(contestResultsModel.Contest.Map<ContestForActivityServiceModel>());
 
         var contestResults = new ContestResultsViewModel
             {
-                Id = contest.Id,
-                Name = contest.Name,
-                IsCompete = official,
+                Id = contestResultsModel.Contest.Id,
```

Comment:
Can you do this using an Automapper configuration?

Reply:
Commenting again so i can requqst changes

Reply:
I think I will need a little help with this

Reply:
Update: Can you check the commit?